core-bundle/src/Resources/contao/library/Contao/User.php

```diff
@@ -509,6 +509,14 @@ public function setUsername($username)
 		return $this;
 	}
 
+	/**
+	 * {@inheritdoc}
+	 */
+	public function getUserIdentifier(): string
+	{
+		return $this->username;
```

Comment:
```suggestion
		return (string) $this;
```

The `toString` method handles empty username cases. We should either use that method or copy it here and use `getUserIdentifier` in `__toString()`.

Reply:
Fixed in 510ca63882b794d4077ec46697608bd4171ed21e